src/nemos/solvers.py

```diff
@@ -0,0 +1,725 @@
+
+        return next_xk
+
+    @partial(jit, static_argnums=(0,))
+    def update(
+        self,
+        current_params: Pytree,
+        state: SVRGState,
+        prox_lambda: Union[float, None],
+        *args,
+    ) -> OptStep:
+        """
+        Perform a single parameter update on the passed data (no random sampling or loops)
+        and increment `state.iter_num`.
+
+        Please note that this is called by `GLM.update`, but repeated calls to `GLM.update`
+        on mini-batches passed to it will not result in running the full (Prox-)SVRG,
+        and parts of the algorithm will have to be implemented outside.
+
+        Parameters
+        ----------
+        current_params :
+            Parameters at the end of the previous update, used as the starting point for the current update.
+        state :
+            Optimizer state at the end of the previous update.
+            Needs to have the current anchor point (xs) and the gradient at the anchor point (df_xs) already set.
+        prox_lambda :
+            Regularization strength.
+        args:
+            Positional arguments passed to loss function `fun` and its gradient (e.g. `fun(current_params, *args)`),
+            most likely input and output data.
+            They are expected to be Pytrees with arrays or FeaturePytree as their leaves, with all of their
+            leaves having the same sized first dimension (corresponding to the number of data points).
+            For GLMs these are:
+                X : DESIGN_INPUT_TYPE
+                    Input data.
+                y : jnp.ndarray
+                    Output data.
+
+
+        Returns
+        -------
+        OptStep
+            xs :
+                Parameters after taking one step defined in the inner loop of Prox-SVRG.
+            state :
+                Updated state.
+
+        Raises
+        ------
+        ValueError
+            The parameter update needs a value for the full gradient at the anchor point, which needs the full data
+            to be calculated and is expected to be stored in state.df_xs. So if state.df_xs is None, a ValueError is raised.
+        """
+        if state.df_xs is None:
+            raise ValueError(
+                "Full gradient at the anchor point (state.df_xs) has to be set. "
+                + "Try passing init_full_gradient=True to ProxSVRG.init_state or GLM.initialize_solver."
+            )
+        return self._update_on_batch(current_params, state, prox_lambda, *args)
+
+    @partial(jit, static_argnums=(0,))
+    def _update_on_batch(
+        self,
+        current_params: Pytree,
+        state: SVRGState,
+        prox_lambda: Union[float, None],
+        *args,
+    ) -> OptStep:
+        """
+        Update parameters given a mini-batch of data and increment iteration/epoch number in state.
+
+        Note that this method doesn't update state.xs, state.df_xs, that has to be done outside.
```

Comment:
would clarify: "e.g., as is done in ..."

Reply:
Do you mean "as is done in `ProxSVRG.run`"? Or how `state.xs` and `state.df_xs` can be updated?

Reply:
I think this is related to the tutorial for `update` that we're going to add. If I understand correctly, the updating "that has to be done outside" isn't currently shown anywhere (except for a test), right? so this will just need to point to that new tutorial that we write

Reply:
yes, I agree

Reply:
I added to the issue on the SVRG tutorial